### PR TITLE
Add `publish` GitHub Actions workflow

### PR DESCRIPTION
**Pull request by nbierdeman**
### Purpose

This PR adds a `publish` GitHub Actions workflow to `paypal-smart-payment-buttons`.

### CODE REVIEW

**Comment by nbierdeman** [comment]
@gregjopa @westeezy, I'm not entirely sure if the `npm run reinstall` step in `publish.yml` will install `devDependencies`, and am therefore not sure if the `npm run build-commit` step in `npm run release` will fail. Do either of you know, or know how to test this prior to merging?

**Review by westeezy — APPROVED** [review]
lgtm to me thanks for adding this. Let's ensure we also update the internal release docs. cc @TNuzzi - I'd be happy to update those if this all looks good to you as well

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+      - name: ⎔ Setup node
+        # sets up the .npmrc file to publish to npm
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node }}
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: 📥 Download deps
+        run: npm run reinstall
```
Comment (gregjopa) [inline .github/workflows/publish.yml:31]: I don't think we need to use the `reinstall` task in SPB. Instead, let's try this:

```
      - name: 📥 Download deps
        uses: bahmutov/npm-install@v1
        with:
          useLockFile: false
```

**Review by gregjopa — APPROVED** [review]
Looks great! 💯

**Pull request merged**